### PR TITLE
[bitnami/keycloak] Add VIB tests

### PR DESCRIPTION
**Pull request by corico44**
### Description of the change

The main objective of this PR is to publish our Bitnami Keycloak container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

### Additional information

The part of code in the build recipe where execution permissions are given to different libraries isn't checked as it depends on the specific architecture being used:
```
    // Some libraries in Keycloak don't have execution permissions and ldd warns about that
    const arch = this.be.target.platform.arch === 'arm64' ? 'aarch64' : 'x86_64';
    const paths = [
      `modules/system/layers/base/org/apache/activemq/artemis/journal/main/lib/linux-${arch}/*`,
      'modules/system/layers/base/org/wildfly/openssl/main/lib/*/*.so',
    ];
    paths.forEach((p) => {
      const libraries = glob.sync(path.join(
        this.prefix,
        p
      ));
      this.logger.debug('Granting execution permisions in the following libraries:', libraries.join(', '));
      libraries.forEach((lib) => fs.chmodSync(lib, 0o754));
    });
  }
```

### CODE REVIEW

File: .vib/keycloak/goss/vars.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+binaries:
+  - java
+  - wait-for-port
```
Comment (FraPazGal) [inline .vib/keycloak/goss/vars.yaml:3]: We could add `kc.sh` here as well.

File: .vib/keycloak/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+directories:
+  - paths:
+      - /opt/bitnami/keycloak/modules/system/layers/keycloak/org/postgresql/jdbc/main
```
Comment (FraPazGal) [inline .vib/keycloak/goss/vars.yaml:6]: Given that you are already checking the `module.xml` file inside this dir, I think we can remove this check.

File: .vib/keycloak/goss/keycloak.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      - "<module xmlns=\"urn:jboss:module:1.0\" name=\"org.postgresql.jdbc\">"
+      - "<resource-root path=\"postgresql-"
+      - "<module name=\"javax.api\"/>"
+      - "<module name=\"javax.transaction.api\"/>"
+command:
+  check-app-version:
+    exec: /opt/bitnami/keycloak/bin/kc.sh --version
+    exit-status: 0          
+    stdout:              
+      - {{ .Env.APP_VERSION }}
```
Comment (FraPazGal) [inline .vib/keycloak/goss/keycloak.yaml:73]: I think we can use the `check-app-version` template using:

```
version:
  bin_name: kc.sh
  flag: --version
```

File: .vib/keycloak/goss/keycloak.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    owner: keycloak
+    filetype: directory
+  /opt/bitnami/keycloak/modules/system/layers/keycloak/org/postgresql/jdbc/main/module.xml:
+    exists: true
+    contains:
+      - "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+      - "<module xmlns=\"urn:jboss:module:1.0\" name=\"org.postgresql.jdbc\">"
+      - "<resource-root path=\"postgresql-"
+      - "<module name=\"javax.api\"/>"
+      - "<module name=\"javax.transaction.api\"/>"
```
Comment (FraPazGal) [inline .vib/keycloak/goss/keycloak.yaml:67]: As we are adding the whole block at the same time after the compilation, we can reduce the `contains` check to just one of these strings, probably one containing a psql mention.

**Review by FraPazGal — COMMENTED** [review]
Nice PR @corico44, I have a added some comments in my review. Please check them whenever possible.

File: .vib/keycloak/goss/keycloak.yaml (hunk 3 above)
Comment (FraPazGal) [reply .vib/keycloak/goss/keycloak.yaml:69]: I think you forgot removing this code 😅 Otherwise, the PR LGTM!

**Review by FraPazGal — COMMENTED** [review]
The changes LGTM @corico44, please ping me once the VIB trigger is removed

**Review by agomezmoron — DISMISSED** [review]

**Review by FraPazGal — APPROVED** [review]

**Pull request merged**